kafka-streams-quickstart/aggregator/src/main/java/org/acme/quarkus/sample/kafkastreams/streams/KafkaStreamsPipeline.java

```diff
@@ -171,25 +178,19 @@ private void waitForTopicsToBeCreated(String bootstrapServers) {
         config.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
 
         try (AdminClient adminClient = AdminClient.create(config)) {
-            AtomicBoolean topicsCreated = new AtomicBoolean(false);
-
-            while (topicsCreated.get() == false) {
+            while (true) {
                 ListTopicsResult topics = adminClient.listTopics();
-                topics.names().whenComplete((t, e) -> {
-                    if (e != null) {
-                        throw new RuntimeException(e);
-                    } else if (t.contains(WEATHER_STATIONS_TOPIC) && t.contains(TEMPERATURE_VALUES_TOPIC)) {
-                        topicsCreated.set(true);
-                    }
-                });
+                Set<String> topicNames = topics.names().get(60, TimeUnit.SECONDS);
 
-                LOG.info("Waiting for topics {} and {} to be created", WEATHER_STATIONS_TOPIC, TEMPERATURE_VALUES_TOPIC);
-                try {
-                    Thread.sleep(1_000);
-                } catch (InterruptedException e) {
-                    throw new RuntimeException(e);
+                if (topicNames.contains(WEATHER_STATIONS_TOPIC) && topicNames.contains(TEMPERATURE_VALUES_TOPIC)) {
+                    return;
                 }
+
+                LOG.info("Waiting for topics {} and {} to be created", WEATHER_STATIONS_TOPIC, TEMPERATURE_VALUES_TOPIC);
+                Thread.sleep(1_000);
```

Comment:
@cescoffier mentioned Awaitility somewhere. Is there a reason why you don't use it? (I might have missed something)

Reply:
So far I've considered Awaitility more as a testing utility (which is why it made sense to me to use it in the integration test). But is it something you'd indeed use in prod code (which more is what this example is supposed to show)? Also it'd mean adding another dependency to the quickstart. But I'm open.

Reply:
I don't have a strong opinion here. Awaitility won't help much in this case and it is a pattern we find a lot when using kafka. So I'm good with the current version.